src/ai/help.rs

```diff
@@ -165,3 +170,33 @@ pub fn prepare_ai_help_summary_req(
 
     Ok(req)
 }
+
+// This function is for QA purposes only, it enables triggering
+// an error based on the input message. The message can be optionally
+// set in the settings `ai.trigger_error_for_search_term`. Nothing
+// will be triggered if the setting is missing, which should be the
+// situation in production-like environments.
+fn qa_check_for_error_trigger(
+    messages: &[ChatCompletionRequestMessage],
+) -> Result<(), async_openai::error::OpenAIError> {
+    if let Some(ref ai) = SETTINGS.ai {
+        if let Some(ref magic_words) = ai.trigger_error_for_search_term {
+            let last_user_message = messages.iter().filter(|m| m.role == Role::User).last();
+            if let Some(msg) = last_user_message {
+                if let Some(ref msg_text) = msg.content {
+                    if msg_text == magic_words {
+                        return Err(async_openai::error::OpenAIError::ApiError(
+                            async_openai::error::ApiError {
+                                message: "Artificial QA error in search phase".to_string(),
+                                r#type: None,
+                                param: None,
+                                code: None,
+                            },
+                        ));
+                    }
+                }
+            }
+        }
+    }
```

Comment:
that's formatted so you can just accept if you think that's _more_ readable.
```suggestion
    if let Some(magic_words) = SETTINGS
        .ai
        .as_ref()
        .and_then(|ai| ai.trigger_error_for_chat_term.as_ref())
    {
        if let Some(msg_text) = messages
            .iter()
            .filter(|m| m.role == Role::User)
            .last()
            .and_then(|m| m.content.as_ref())
        {
            if msg_text == magic_words {
                return Err(async_openai::error::OpenAIError::ApiError(
                    async_openai::error::ApiError {
                        message: "Artificial QA error in search phase".to_string(),
                        r#type: None,
                        param: None,
                        code: None,
                    },
                ));
            }
        }
    }
```